### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by rcknight**
See Readme.md for details and usage

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    using LibGit2Sharp;
+    using Microsoft.Extensions.CommandLineUtils;
+
+    public static class Program
+    {
+        public static void Main(string[] args)
+        {
+            var app = new CommandLineApplication();
+            app.Name = "MassPullRequest";
+            app.Description = "Clones then executes an action accross a number of GitHub repositories, optionally committing and creatign a PR with any changes";
```
Comment (jbradford) [inline Program.cs]:
```suggestion
            app.Description = "Clones then executes an action accross a number of GitHub repositories, optionally committing and creating a PR with any changes";
```

File: Program.cs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+            app.HelpOption("-?|-h|--help");
+
+            var command = app.Argument("Command", "The command to be executed on each repostitory, executed via powershell on windows, bash elsewhere.", false);
```
Comment (jbradford) [inline Program.cs]:
```suggestion
            var command = app.Argument("Command", "The command to be executed on each repository, executed via powershell on windows, bash elsewhere.", false);
```

File: Program.cs [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+                CommandOptionType.SingleValue);
+
+            var doCommit = app.Option(
+                "--commit",
+                "Set this flag to create a commit with changes to the repository. If using this flag, a commit message must be set with --commit-message.",
+                CommandOptionType.NoValue);
+
+            var doPush = app.Option(
+                "--push",
+                "Set this flag to push changes to the remote repository.",
```
Comment (jbradford) [inline Program.cs]:
```suggestion
                "Set this flag to push changes to the remote repository. Implies --commit.",
```

File: Program.cs [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+                    if (createPr)
+                    {
+                        processor.CreatePullRequest(commitMessage);
+                    }
+                }
+                finally
+                {
+                    // Wether success or failure, clean up the cloned repository
+                    if (processor != null)
```
Comment (jbradford) [inline Program.cs]: Shouldn't this check the `cleanup` parameter?

File: Program.cs [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+            var credentials = new UsernamePasswordCredentials() { Username = userName, Password = password };
+            return credentials;
+        }
+
+        private static string GetHiddenConsoleInput()
+        {
+            StringBuilder input = new StringBuilder();
+            while (true)
+            {
+                var key = Console.ReadKey(true);
```
Comment (jbradford) [inline Program.cs:194]: If we're adding/removing by character wouldn't this be a perfect candidate for using a SecureString https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=netcore-2.2

Comment (jbradford) [reply Program.cs:194]: May not be possible if other libraries don't support this.

File: Program.cs [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+            }
+
+            return 0;
+        }
+
+        private static UsernamePasswordCredentials GetCredentials()
+        {
+            Console.Write("Enter git username: ");
+            var userName = Console.ReadLine();
+            Console.Write("Enter git password: ");
```
Comment (jbradford) [inline Program.cs]: May be worth mentioning this can be an access key (and must be for 2FA users).

**Review by jbradford — CHANGES_REQUESTED** [review]
A few tweaks + small bug.

File: Program.cs (hunk 5 above)
Comment (rcknight) [reply Program.cs:194]: Libgit2sharp does not support secure strings unfortunately.

**Review by jbradford — APPROVED** [review]

**Pull request merged**